0-common/igluctl/src/main/scala/com.snowplowanalytics.iglu/ctl/commands/Pull.scala

```diff
@@ -92,18 +96,24 @@ object Pull {
       else EitherT.leftT[IO, HttpResponse[String]](s"Unexpected status code ${response.code}. Response body: ${response.body}.")
     }
 
-  /**
-    * Parse the response from Iglu Server and write it under the specified output path.
-    * Perorms IO
+  /** Parse the response from Iglu Server
     *
     * @param response An [[HttpResponse]] returned by Iglu Server
+    * @return A list of [[SelfDescribingSchema]]
+    */
+  def parseResponse(response: HttpResponse[String]): List[Either[IgluParseError, SelfDescribingSchema[JValue]]] =
+    jacksonParse(response.body).extract[List[JValue]].map(igluParse[JValue](_))
```

Comment:
`extract` is still impure method that throws an exception in case `HttpResponse` contains invalid JSON.

But bigger point - I'm wondering if we *list of invalid and valid schemas* is what we really need here. I think if any of schemas is invalid - there's something wrong with this response. This is however just my UX opinion - however if you're agree with it the return type will `Either[ParseError, List[SelfDescribingSchema[JValue]]]`, which means response can be either *error or list of valid schemas*. Also, you're doing `sequence` in `writeSchemas`, so in fact you still get the type I mentioned, so why not to do it here?

Also I personally believe that `IgluParseError` can be replaced with `Common.Error` at this moment. It is a general error type for igluctl and what's the point of doing this transformation in `writeSchemas`.